nbextensions/appCell2/widgets/tabs/jobStateListRow.js

```diff
@@ -71,156 +64,64 @@ define([
-            case 'suspend':
-            case 'canceled':
-                stopJobUpdates();
-                break;
-            default:
-                stopJobUpdates();
-                console.error('Unknown job status', jobState.job_state, message);
-                throw new Error('Unknown job status ' + jobState.job_state);
-            }
-        }
-
-        function listenForJobStatus() {
-            var ev = runtime.bus().listen({
-                channel: {
-                    jobId: jobId
-                },
-                key: {
-                    type: 'job-status'
-                },
-                handle: handleJobStatusUpdate
-            });
-            listeners.push(ev);
-
-            ev = runtime.bus().listen({
-                channel: {
-                    jobId: jobId
-                },
-                key: {
-                    type: 'job-canceled'
-                },
-                handle: function() {
-                    console.warn('job cancelled');
-                }
-            });
-            listeners.push(ev);
-
-            ev = runtime.bus().listen({
-                channel: {
-                    jobId: jobId
-                },
-                key: {
-                    type: 'job-does-not-exist'
-                },
-                handle: handleJobDoesNotExistUpdate
-            });
-            listeners.push(ev);
-        }
-
-        function stopListeningForJobStatus() {
-            runtime.bus().removeListeners(listeners);
+            container.innerHTML = th({}, [div(isParentJob ? name.toUpperCase() : name), jobIdDiv]) + niceState(jobStatus);
         }
 
         function start(arg) {
             return Promise.try(function() {
-                container = arg.node;
+                container = arg.node;               // this is the row (tr) that this renders
+                container.onclick = () => {
+                    if (jobId) {
+                        clickFunction(container, jobId, isParentJob);
+                    }
+                };
                 ui = UI.make({ node: container });
 
-                jobId = arg.jobId;
-                var clickFunction = arg.clickFunction;
-                listeners.push(runtime.bus().on('clock-tick', function() {
-                    updateRowStatus(ui, jobState, container, clickFunction);
-                }));
-
-                listenForJobStatus();
-                runtime.bus().emit('request-job-status', {
-                    jobId: jobId
-                });
-
-                listeningForJob = true;
-
+                jobId = arg.jobId;                  // id of child job
+                name = arg.name;
+                isParentJob = arg.isParentJob;
+                clickFunction = arg.clickFunction;  // called on click (after some ui junk)
+                updateRowStatus(arg.initialState);
             });
         }
 
         function stop() {
-            return Promise.try(function() {
```

Comment:
why is stop empty?

Reply:
Related to below, it no longer does anything that needs to stop, but the stop function still needs to be present to be a valid widget.

Reply:
clickFunction starts a widget?

Reply:
Yes, indirectly. It toggles between what child jobs are being displayed. There's 3 display widgets - Job Params, Job Status, and Job Log. Clicking on a different child job calls `stop()` on each of those, then `start(jobId)` on the new jobId (ok, that's not the real syntax, but you get it).

Clicking on 'View Configure' tells everything under the 'Job Status' tab to stop, which includes all of those widgets as well.

Reply:
ohh okay so basically the row doesn't have anything it needs to stop as the child job display widgets stop and restart with the clicked jobId.

Reply:
Yup, exactly. It's now just a little thing that formats and displays whatever state gets pushed to it from jobStateList.